### ISSUE TITLE
zfsUnstable rc5 fails to build

### ISSUE THREAD

**Issue by luigy** [NixOS/nixpkgs#27661]
cc @Mic92 (seems familiar with this)

maybe patches need to be updated for rc5?

## Issue description

```
  CC [M]  /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zcommon/zfs_fletcher_avx512.o
  LD [M]  /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zcommon/zcommon.o
  CC [M]  /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zfs/abd.o
  CC [M]  /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zfs/arc.o
In file included from /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/sys/zpl.h:37:0,
                 from /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zfs/arc.c:281:
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h: In function 'zpl_bdi_setup':
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h:154:22: error: 'BDI_CAP_MAP_COPY' undeclared (first use in this function)
  bdi->capabilities = BDI_CAP_MAP_COPY;
                      ^
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h:154:22: note: each undeclared identifier is reported only once for each function it appears in
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h:156:10: error: implicit declaration of function 'bdi_init' [-Werror=implicit-function-declaration]
  error = bdi_init(bdi);
          ^
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h:162:34: warning: too many arguments for format [-Wformat-extra-args]
  error = bdi_register(bdi, NULL, "%.28s-%ld", name,
                                  ^
/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/include/linux/vfs_compat.h:165:3: error: implicit declaration of function 'bdi_destroy' [-Werror=implicit-function-declaration]
   bdi_destroy(bdi);
   ^
cc1: some warnings being treated as errors
make[7]: *** [/nix/store/ix8if9li2p14if8sky007lipg0mydpni-linux-4.12.3-dev/lib/modules/4.12.3/source/scripts/Makefile.build:303: /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zfs/arc.o] Error 1
make[6]: *** [/nix/store/ix8if9li2p14if8sky007lipg0mydpni-linux-4.12.3-dev/lib/modules/4.12.3/source/scripts/Makefile.build:561: /tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module/zfs] Error 2
make[5]: *** [/nix/store/ix8if9li2p14if8sky007lipg0mydpni-linux-4.12.3-dev/lib/modules/4.12.3/source/Makefile:1512: _module_/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module] Error 2
make[4]: *** [Makefile:152: sub-make] Error 2
make[3]: *** [Makefile:24: __sub-make] Error 2
make[3]: Leaving directory '/nix/store/ix8if9li2p14if8sky007lipg0mydpni-linux-4.12.3-dev/lib/modules/4.12.3/build'
make[2]: *** [Makefile:29: modules] Error 2
make[2]: Leaving directory '/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src/module'
make[1]: *** [Makefile:720: all-recursive] Error 1
make[1]: Leaving directory '/tmp/nix-build-zfs-kernel-0.7.0-rc5-4.12.3.drv-0/zfs-zfs-0.7.0-rc5-src'
make: *** [Makefile:589: all] Error 2
builder for ‘/nix/store/hlnzjm66s6smqs63p35l018f7b838bm7-zfs-kernel-0.7.0-rc5-4.12.3.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/jmcyi92cwg9zgd3l8pf7p15m3y4qj962-kernel-modules.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ssixfl4w6408976ld252135abidj6dyf-nixos-system-bolon-17.09pre111388.00512470ec.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/ssixfl4w6408976ld252135abidj6dyf-nixos-system-bolon-17.09pre111388.00512470ec.drv’ failed

```

### Steps to reproduce
build `zfsUnstable`

## Technical details

* System: NixOS: 17.09pre110213.01c3847b9c
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: 17.09pre111388.00512470ec
* Sandboxing enabled: true

**Comment by Mic92** [comment]
Yeah, you need to temporary downgrade the kernel from 4.12. There is one bug in the buildsystem, which leads to newer code not beeing selected and it falls back to incompatible one. I have not yet figured out why the compile tests in the configure script fails.

**Comment by luigy** [comment]
Thanks, indeed `0.7.0` builds for me. 

Fixed by 2799a94963aaf37f059b5ed4c0d2b0cf98ba445e
